src/libraries/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/SignedXml.cs

```diff
@@ -100,7 +101,7 @@ public SignedXml(XmlElement elem!!)
             Initialize(elem);
         }
 
-        private void Initialize(XmlElement element)
+        private void Initialize(XmlElement? element)
```

Comment:
```suggestion
        [MemberNotNull(nameof(_containingDocument))]
        [MemberNotNull(nameof(m_signature))]
        [MemberNotNull(nameof(_safeCanonicalizationMethods))]
        private void Initialize(XmlElement? element)
```

And then `m_signature` (and others) can be marked as non-nullable, and `!`s can be removed.

Reply:
Great suggestion, thanks! Those attributes are great.  However, it doesn't work for `_containingDocument` because:
```csharp
        private void Initialize(XmlElement? element)
        {
            _containingDocument = (element == null ? null : element.OwnerDocument);
```

So, `_containingDocument` can be `null` of `element` is null.